### PR TITLE
feat(Validation): add domain validator

### LINKED ISSUE #1235: Validator: Domain validator
Currently, core has a need to validate domains but not full URLs in some cases.

Leverage a domain regex from the tools library.

- [ ] Create validator
- [ ] Add to demo
- [ ] Add to usage docs as needed
- [ ] Add to primary readme
- [ ] Add to demo

### PR DESCRIPTION
**Pull request by atlwendy**
closes #1235

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -144,7 +144,7 @@
     "@angular/platform-browser": "^7.2.2",
     "@angular/platform-browser-dynamic": "^7.2.2",
     "@angular/router": "^7.2.2",
-    "@terminus/ngx-tools": "latest",
+    "@terminus/ngx-tools": "^5.2.0",
```
Comment (shani-terminus) [inline package.json]: This should be "latest"
I understand their were some issues last week, but we now have a work-around.

Comment (benjamincharity) [reply package.json]: Once it's changed back to `latest` we should be able to run `yarn upgrade @terminus/ngx-tools` to update the lock file so that `5.2.0` is installed

File: README.md [hunk 2]
```diff
@@ -165,6 +165,7 @@ for each feature.
 | Name             | Purpose                                                       | Status             |
 |------------------|---------------------------------------------------------------|:------------------:|
 | `creditCard`     | A credit card number must be valid                            | :white_check_mark: |
+| `domain`         | An domain  must be valid                                      | :white_check_mark: |
```
Comment (benjamincharity) [inline README.md]: Spelling `An` vs `A`. Plus extra space after `domain`

File: demo/app/components/validation/validation.component.html [hunk 3]
```diff
@@ -197,28 +197,28 @@ <h3 tsCardTitle tsVerticalSpacing="small--0">Contains numbers</h3>
...
-// Invalid URLs:
-http://
-http://foo.bar?q=Spaces should be encoded
-foo.com
-rdar://1234
-http://1.1.1.1.1
-http://.www.foo.bar./
-http:// shouldfail.com
-</pre>
+      <pre fxFlex>
```
Comment (benjamincharity) [inline demo/app/components/validation/validation.component.html]: We should undo this change.. I know it makes indentation super odd, but the `pre` element will always respect indentation. So this will actually output 3 indentions in the UI and we really want it to be at 0 indentation when output.

File: terminus-ui/validators/src/validators/domain/domain.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import {
+  ValidatorFn,
+  AbstractControl,
+  ValidationErrors,
```
Comment (benjamincharity) [inline terminus-ui/validators/src/validators/domain/domain.ts]: Let's sort these while we're in here
